### DISCUSSION TITLE
Help Connecting to Self-Hosted Supabase Database in DataGrip

### DISCUSSION THREAD

**Discussion by InvixGG** [supabase/supabase#30209] [Questions]
Hello everyone,

I'm having trouble connecting to my self-hosted Supabase PostgreSQL database using JetBrains DataGrip. I've set up Supabase in a Docker Compose environment, exposing the database on port `5434`. However, when I attempt to connect through DataGrip, I encounter connection errors.

Here's a summary of my setup and the errors I've encountered:
### Environment Details
- **PostgreSQL Version**: 15.1
- **Database User**: `supabase_admin`
- **Database Password**: (using environment variables)
- **OS**: Windows 11

### Connection Setup in DataGrip
- **Host**: `localhost`
- **Port**: `5434`
- **Database**: `postgres`
- **User**: `supabase_admin`
- **Authentication**: User & Password

### Errors
When I try to connect, I see the following error message:
- **Error**: `Connection to localhost:5434 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.`

### What I've Tried
1. Confirmed that the `.env` file is correctly loaded with the right port (`5434`).
2. Verified that the PostgreSQL service is running and mapped to `5434`.

Any help or guidance would be greatly appreciated! Thank you in advance!

**Comment by CodingMeSwiftly** [comment]
Works for me, actually. See the screenshot below. I didn't change any ports or env variables, just plain `docker pull`.
I had to create a new datasource though. Updating my existing one in DataGrip which pointed to my local postgres before did not work.

## Old comment

I got a bit further using these connection settings. Password: `your-super-secret-and-long-postgres-password`

<img width="1194" height="947" alt="Screenshot 2025-11-13 at 17 15 57" src="https://github.com/user-attachments/assets/5124f31d-25ad-4291-a71e-5416d7e753a5" />

`Test connection` shows success, but trying to access the connection after `Apply` causes a timeout after 10s with this message:

`An I/O error occurred while sending to the backend.`

I haven't tried any other psql client or using direct connection via port `5432` (requires disabling Supavisor).
